test/functional/zmq_test.py

```diff
@@ -13,9 +13,33 @@
                                  hash256,
                                 )
 
+class ZMQSubscriber:
+    def __init__(self, socket, address):
+        self.address = address
+        self.socket = socket
+        self.sequence = 0
+
 class ZMQTest (BitcoinTestFramework):
     def set_test_params(self):
+        self.address = "tcp://127.0.0.1:28332"
+        self.num_blocks = 5
         self.num_nodes = 2
+        self.subscribers = {}
+
+    def subscribe(self, type):
+        import zmq
+        self.socket.setsockopt(zmq.SUBSCRIBE, type.encode('latin-1'))
+        self.subscribers[type] = ZMQSubscriber(self.socket, self.address)
+
+    def receive(self, type):
+        subscriber = self.subscribers[type]
+        topic, body, seq = subscriber.socket.recv_multipart()
```

Comment:
The subscribers don't actually have independent sockets, perhaps just use self.socket?

Reply:
I would like to leave it this way so in the future we can have more sockets.